packages/shared/src/models/dataPipe/operators/filter.ts

```diff
@@ -39,6 +39,20 @@ class FilterOperator extends BaseOperator {
       }
     });
   }
+
+  /**
+   * Create a list of all key-value pairs of row data whose keys appear within condition string
+   * This reduces the complexity of later JS evaluation
+   **/
+  private generateEvaluatorConstants(condition: string, row: Record<string, any>) {
+    const constants: Record<string, any> = {};
+    for (const [key, value] of Object.entries(row)) {
+      if (condition.includes(key)) {
```

Comment:
This could lead to some key-value pairs being erroneously included, for example if the condition refers to a "name" column and the row data includes a "first_name" column. Seeing as the purpose of this step is efficiency/reducing complexity of the later JS evaluation, I don't think that's a problem, as it would still be an improvement in such cases.

Reply:
I think in that case if the condition were `name==='Ada'` then `condition.includes('first_name')` would still return false and not be included.

Although yes the opposite case would be true, `first_name==='Ada'` and `condition.includes('name')`.
Given that the variable can be used within any valid javascript I think the regex gets quite tricky to pin down perfectly (would need regex that allows any valid javascript syntax/operators either side of the condition text but not anything that could be included as part of a variable name - I'm assuming these are actually distinct sets given variable naming restrictions, however I'm less confident writing the regex to do so)

Reply:
A couple sources point in various directions to do so
https://mathiasbynens.be/demo/javascript-identifier-regex
https://regex101.com/r/7fjduD/1

However given the likely complexity I think I'd be much more in favor of accepting the limitation, and if desired exposing a config option to operations like `filter` that allow the user to explicitly pass names of columns to include for advanced use